dns/dns_metrics.go

```diff
@@ -13,42 +13,45 @@ var updateCount, failedCount, skippedCount prometheus.Counter
 
 func initMetrics() {
 	once.Do(func() {
-		recordsGauge = prometheus.MustRegisterOrGet(prometheus.NewGauge(
+		recordsGauge = prometheus.NewGauge(
 			prometheus.GaugeOpts{
 				Namespace:   metrics.PrometheusNamespace,
 				Subsystem:   metrics.PrometheusDNSSubsystem,
 				Name:        "route53_records",
 				Help:        "The current number of records.",
 				ConstLabels: metrics.ConstLabels(),
-			})).(prometheus.Gauge)
+			})
+		prometheus.MustRegister(recordsGauge)
```

Comment:
There's quite a lot of repetition for this function call. Can we either use `prometheus.Register()` to register a single metric, or prepare a slice of `prometheus.Collector` to call `prometheus.MustRegister()` once with?

Reply:
Most collectors use default options (`namespace`, `subsystem`, `...`) so I have introduced a helper function `prometheus.NewDefaultGauge()` to create them (see https://github.com/sky-uk/feed/pull/207/commits/42fb04086574d3c5aafef473692225ba3936b485)